demo_scripts/test/README.md

```diff
@@ -3,6 +3,36 @@
 This is a simple test script. It runs a number of commands in
 succession. This script also lists commands known not to work.
 
+## Basic Code block and results section
+
+Ensure things work with a bash code block in various formats.
+
+```bash
+# This is a code block, this comment will be ignored by SimDem
+echo "This command will be 'typed' and executed."
+```
+
+Code blocks can optionally be indented to make the markdown more 
+readable. For example:
+
+    ```
+    echo "Hello world indent"
+    ```
+
+Results:
+```expected_similarity=0.6
+Hello world indent
+```
+
+If a code block exists with a different language hint, it will not be
+execute, but instead will be output as formatted code, for example:
+
+```json
+{
+  Foo: "bar"
+}
+```
+
```

Comment:
It appears that the break here is anytime there is text after ```. I did some debugging and when I removed both the json and indent examples the program exited immediately after echo $SIMDEM_VERSION. The reason is because ```expected_similarity=0.4 is after. 

When I removed this line, the next line in the test it broke at was head -n 1 README.md. This is also because it has an ```expected_similarity=0.8 after 

The indent code appears to be working correctly, but it exits as expected_similarity is after that. 

Similarly, the program exits immediately after the ```json block.

Reply:
All 21 tests in the current test suite now pass.